azure-storage/spring-cloud-starter-stream-sink-azure-blob/src/main/java/org/springframework/cloud/stream/app/azure/storage/sink/AzureBlobSinkConfiguration.java

```diff
@@ -0,0 +1,140 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.azure.storage.sink;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessagingException;
+
+// Include the following imports to use blob APIs.
+import com.microsoft.azure.storage.CloudStorageAccount;
+import com.microsoft.azure.storage.blob.BlobContainerPermissions;
+import com.microsoft.azure.storage.blob.BlobContainerPublicAccessType;
+import com.microsoft.azure.storage.blob.CloudBlob;
+import com.microsoft.azure.storage.blob.CloudAppendBlob;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+
+
+/**
+ * @author Kyle Dunn
+ */
+@EnableBinding(Sink.class)
+@EnableConfigurationProperties(AzureBlobSinkProperties.class)
+@SpringBootApplication
+public class AzureBlobSinkConfiguration {
+
+    protected static Logger LOG = LoggerFactory.getLogger(AzureBlobSinkConfiguration.class);
+
+    @Autowired
+    private AzureBlobSinkProperties properties;
+
+    private CloudBlob blobService;
+
+    @Autowired
+    public void setBlobService() {
+        // Define the connection-string with your values
+        final String storageConnectionString =
+            "DefaultEndpointsProtocol=" + this.properties.getDefaultEndpointsProtocol() +
+            ";AccountName=" + this.properties.getAccountName() +
+            ";AccountKey=" + this.properties.getAccountKey();
+
+        try {
+            // Setup the cloud storage account.
+            CloudStorageAccount account = CloudStorageAccount.parse(storageConnectionString);
+
+            LOG.info("getBlobService() : using account {}", this.properties.getAccountName());
+
+            // Create a blob service client
+            CloudBlobClient blobClient = account.createCloudBlobClient();
+            
+            // Get a reference to a container
+            // The container name must be lower case
+            CloudBlobContainer container = blobClient.getContainerReference(this.properties.getContainerName().toLowerCase());
```

Comment:
A blob container can be as an option of incoming message, e.g. some header.
Therefore any reference to `CloudBlobContainer` object should be done during `@ServiceActivator` process.

And in general it is would be much better do not do any hard network processes during bean initialization phase.
For this purpose we have `SmartLifecycle` contract with its `autoStartup = true`.

Looks like `CloudStorageAccount` and `CloudBlobClient` instantiations are safe of network operations. Therefore we still can do that during `@PostConstruct` phase. Although, as I said before, it would be better to move the to separate `@Configuration`